.github/actions/common/init-and-pull-gcc/action.yaml

```
@@ -49,3 +55,28 @@ runs:
           git checkout master
           git pull
 
+      - name: Checkout branch
+        shell: bash
+        if: ${{ inputs.branch != 'master' }}
+        working-directory: ./riscv-gnu-toolchain
+        id: pull
+        run: |
+          cd gcc
+          git config --add remote.origin.fetch "+refs/vendors/riscv/*:refs/remotes/upstream/vendors/riscv/*"
+          git fetch
+          git checkout upstream/vendors/riscv/heads/gcc-14-with-riscv-opts
+        continue-on-error: true
+
+      - name: Sleep and retry
+        shell: bash
+        if: ${{ inputs.branch != 'master' }}
+        working-directory: ./riscv-gnu-toolchain
+        if: ${{ steps.pull.outcome == 'failure' }}
+        run: |
+          echo "Failed to checkout and pull gcc. Retrying in 1 min"
+          sleep 60
+          cd gcc
+          git config --add remote.origin.fetch "+refs/vendors/riscv/*:refs/remotes/upstream/vendors/riscv/*"
+          git fetch
+          git checkout upstream/vendors/riscv/heads/gcc-14-with-riscv-opts
+
```

re: making a positive check rather than a negative check.
```suggestion

      - name: Detect unrecognized branches
        shell: bash
        if: ${{ inputs.branch != 'master' && inputs.branch != 'coordination' }}
        run: exit 1
```